nucypher/blockchain/eth/utils.py

```diff
@@ -162,78 +160,22 @@ def get_default_rpc_endpoints() -> Dict[int, List[str]]:
         return {}
 
 
-def get_healthy_default_rpc_endpoints(chain_id: int) -> List[str]:
+def get_healthy_default_rpc_endpoints(domain: str) -> Dict[int, List[str]]:
     """Returns a list of healthy RPC endpoints for a given chain ID."""
 
-    endpoints = get_default_rpc_endpoints()
-    chain_endpoints = endpoints.get(chain_id)
-
-    if not chain_endpoints:
-        LOGGER.error(f"No default RPC endpoints found for chain ID {chain_id}")
-        return list()
-
-    healthy = [
-        endpoint for endpoint in chain_endpoints if rpc_endpoint_health_check(endpoint)
-    ]
-    LOGGER.info(f"Healthy default RPC endpoints for chain ID {chain_id}: {healthy}")
-    if not healthy:
-        LOGGER.warn(
-            f"No healthy default RPC endpoints available for chain ID {chain_id}"
-        )
+    endpoints = get_default_rpc_endpoints(domain)
+
+    if domain == "mainnet":
+        # iterate over all chains and filter out unhealthy endpoints
+        healthy = {
```

Comment:
```suggestion
        # iterate over all chains and filter out unhealthy endpoints
        # only performed for mainnet for now since it would be inefficient 
        #  on testnet where many more chains are/will be allowed
        healthy = {
```